### DISCUSSION TITLE
DAX Client

### DISCUSSION THREAD

**Discussion by mrvini** [sensedeep/dynamodb-onetable#309] [Q&A] [answered]
Hi everyone, 

Did anyone tried to use Dax Client or was able to connect to DAX cluster with the dynamodb-onetable library?
What we'e tried is to switch endpoint to point to `dax://...` but that didn't work, when sending dax client instead of dynamodb client, that also didn't seem to work

any suggestions?

**Comment by mobsense** [comment]
I've not tried that. Can you post the sample of what you are trying?

You must provide a DocumentClient instance to onetable and not the Dax instance.

**Comment by marcosbasualdo** [comment, marked as answer]
We were able to solve it. Apparentely, when using **AWS SDK V3** DAX is not supported by the official DynamoDBClient package. And the way of initializing a client to be used by dynamodb-onetable is **wrapping the V3 client with the Dynamo adapter from dynamodb-onetable**:

``` javascript
import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import Dynamo from 'dynamodb-onetable/Dynamo'

export const client = new Dynamo({
    client: new DynamoDBClient({
        region: process.env.AWS_REGION
    })
});
```

In order to use DAX, we had to use **AWS SDK V2**, and this version **should not be wrapped with the dynamodb-onetable Dynamo adapter**, the V2 client should be used directly by dynamodb-onetable schemas. And the initialization code with DAX looks as follows:

``` javascript
import { DynamoDB } from 'aws-sdk';
import AmazonDaxClient from 'amazon-dax-client'

const endpoint = "dax://DAX-CLUSTER-ENDPOINT";
const dax = new AmazonDaxClient({endpoints: [endpoint], region: 'AWS_REGION'});

export const client = new DynamoDB.DocumentClient({ service: dax });
```

**Reply by mrvini** [reply]
awesome, it'd be good to go in the readme

**Reply by mobsense** [reply]
Will add to the doc today.

**Reply by mobsense** [reply]
Here is the link. Let me know if I've got this right:

https://doc.onetable.io/api/table/constructor/constructor/#aws-dynamodb-accelerator-dax

**Reply by marcosbasualdo** [reply]
You got it right, thanks!!
